src/app/features/container/headers/header.tsx

```diff
@@ -0,0 +1,83 @@
+import { useState } from 'react';
+import { useNavigate } from 'react-router-dom';
+
+import { ChainID } from '@stacks/transactions';
+import { OnboardingSelectors } from '@tests/selectors/onboarding.selectors';
+import { SettingsSelectors } from '@tests/selectors/settings.selectors';
+import { SharedComponentsSelectors } from '@tests/selectors/shared-component.selectors';
+import { Box, Flex, Grid, GridItem, HStack, styled } from 'leather-styles/jsx';
+
+import { ArrowLeftIcon, HamburgerIcon, Logo, NetworkModeBadge } from '@leather.io/ui';
+
+import { RouteUrls } from '@shared/route-urls';
+
+import { SwitchAccountDialog } from '@app/features/dialogs/switch-account-dialog/switch-account-dialog';
+import { Settings } from '@app/features/settings/settings';
+import { useCurrentNetworkState } from '@app/store/networks/networks.hooks';
+import { HeaderActionButton } from '@app/ui/layout/containers/headers/components/header-action-button';
+
+interface HeaderProps {
+  hideBackButton?: boolean;
+  hideLogo?: boolean;
+}
+
+export function Header({ hideBackButton = false, hideLogo = false }: HeaderProps) {
+  const { chain, name: chainName } = useCurrentNetworkState();
+  const [isShowingSwitchAccount, setIsShowingSwitchAccount] = useState(false);
+
+  const navigate = useNavigate();
+  return (
+    <>
```

Comment:
I need to do some refactors of `Header` files to make the code more DRY and share the layout